pwndbg/heap/ptmalloc.py

```diff
@@ -53,38 +52,70 @@ def mp(self):
 
         return self._mp
 
+
     @property
     def malloc_chunk(self):
         if not self._malloc_chunk:
             self._malloc_chunk = pwndbg.typeinfo.load('struct malloc_chunk')
         return self._malloc_chunk
 
+
     @property
     def malloc_state(self):
         if not self._malloc_state:
             self._malloc_state = pwndbg.typeinfo.load('struct malloc_state')
         return self._malloc_state
 
+
     @property
     def mallinfo(self):
         if not self._mallinfo:
             self._mallinfo = pwndbg.typeinfo.load('struct mallinfo')
         return self._mallinfo
 
+
     @property
     def malloc_par(self):
         if not self._malloc_par:
             self._malloc_par = pwndbg.typeinfo.load('struct malloc_par')
         return self._malloc_par
 
+    def _spaces_table(self):
+        spaces_table =  [ pwndbg.arch.ptrsize * 2 ]      * 64 \
+                      + [ pow(2, 6) ]                    * 32 \
+                      + [ pow(2, 9) ]                    * 16 \
+                      + [ pow(2, 12) ]                   * 8  \
+                      + [ pow(2, 15) ]                   * 4  \
+                      + [ pow(2, 18) ]                   * 2  \
+                      + [ pow(2, 21) ]                   * 1
+
+        # There is no index 0
+        spaces_table = [ None ] + spaces_table
+
+        # Fix up the slop in bin spacing ( part of libc - they made
```

Comment:
I don't believe the bin design has changed at all since 2000.  It uses the same bin design as the original `dlmalloc`.

It looks like `dlmalloc` 2.6.6 was the first version to have exactly that macro (`bin_at`) with that description:
http://g.oswego.edu/pub/misc/malloc-2.6.6.c

Since Ubuntu is our target, I'd rather link directly to their release, as far as "linking to documentation" goes.  For example:

https://bazaar.launchpad.net/~ubuntu-branches/ubuntu/trusty/eglibc/trusty-security/view/head:/malloc/malloc.c#L1356